### ISSUE TITLE
LLVM ERROR about f128 on AIX and SPARC32 when building core without optimization

### ISSUE THREAD

**Issue by taiki-e** [rust-lang/rust#133035]
### Code

I tried this code:

```rust
#![no_std]
// empty
```

With `cargo build -Z build-std=core --target powerpc64-ibm-aix`:

```
rustc-LLVM ERROR: f128 is unimplemented on AIX.
error: could not compile `core` (lib)
```

With `cargo build -Z build-std=core --target sparc-unknown-none-elf`:

```
rustc-LLVM ERROR: SPARCv8 does not handle f128 in calls; pass indirectly
error: could not compile `core` (lib)
```

I expected to see this happen: no error

I suspect the same problem exists with the other targets listed here: https://github.com/rust-lang/compiler-builtins/blob/a09218f1c4d23ffbd97d68f0fefb5feed2469dc5/configure.rs#L82-L91

Workaround: Passing `--release` flag resolves error.

### Version it worked on

It most recently worked on: nightly-2024-11-13

### Version with regression

nightly-2024-11-14

`rustc --version --verbose`:
```
rustc 1.84.0-nightly (8adb4b30f 2024-11-13)
binary: rustc
commit-hash: 8adb4b30f40e6fbd21dc1ba26c3301c7eeb6de3c
commit-date: 2024-11-13
host: aarch64-apple-darwin
release: 1.84.0-nightly
LLVM version: 19.1.3
```

@rustbot label +regression-from-stable-to-nightly -regression-untriaged

**Comment by taiki-e** [comment]
Considering [this comment](https://github.com/rust-lang/rust/pull/132662/files#diff-21f48a7ba852bc22e2b05761c69006a742522cef67d1acedffae71babf6366c3R1271), I suspect it is a regression introduced by https://github.com/rust-lang/rust/pull/132662.
cc @tgross35 @RalfJung

**Comment by taiki-e** [comment]
> I suspect it is a regression introduced by [#132662](https://github.com/rust-lang/rust/pull/132662).

I confirmed that reverting this PR's change fixes the problem locally.

**Comment by workingjubilee** [comment]
`-Zbuild-std` without optimizations often fails. We should not really consider it even an experimentally supported use-case until we fix that.
